### ISSUE TITLE
Cannot set YamlMemberAttribute.Order as an additional argument

### ISSUE THREAD

**Issue by mvantzet** [xoofx/SharpYaml#42]
First of all thanks for SharpYaml! If I want to set both the name and the order for serializing a property like this:

```cs
[YamlMember("someprop", Order = 1)]
public string SomeProperty { get; set; }
```

I get a compiler error:

"An attribute argument must be a constant expression, typeof expression or array creation expression of an attribute parameter type." See also this SO answer that references the spec: https://stackoverflow.com/a/3192882

The solution is to either create an overload for the `YamlMemberAttribute` constructor so that you can set both name and order, or to change the type of `Order` from "int?" to "int". The last solution is the most flexible but I don't know what the consequences are of changing `Order`'s type (i.e. if null has a special meaning).

**Comment by LodrikMtl** [comment]
I'm still having this issue! Is there any improvement?

**Comment by 2bam** [comment]
I have the same issue (.NET 4.7.1  VS2017 v15.9.26), it happens because `Order` can't be a custom attribute parameter because it's nullable `int?` and by definition can't be a constant literal which is the only type attribute parameters can take.

Found the explanation here: 
https://stackoverflow.com/questions/1416126/why-is-a-nullablet-not-a-valid-custom-attribute-parameter-when-t-is

If either `Name` had a setter or `Order` was a normal `int` it could be worked around.

**Comment by jbienz** [comment]
@xoofx could you please add a constructor overload that takes both **name** and **order**? As others have pointed out, int? is not a valid attribute parameter type which results in a [CS0655 compiler error](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0655).

My application is editing configuration files for an [EdgeTX radio](https://edgetx.org/about). I want to load properties by name in case they're out of order, but when I write them back out I'd like to write them in the same sequence that the radio would.

**Comment by jbienz** [comment]
Alternatively, I see this comment on the **Order** property:

> Gets or sets the order. Default is -1 (default to alphabetical)

If the order does indeed default to -1, this property could be changed from int? to int and that would resolve the CS0655 error.

**Comment by xoofx** [comment]
PR welcome

**Comment by jbienz** [comment]
Actually, it appears that comment is incorrect. It doesn't default to -1 and there is code that explicitly checks for its nullness (HasValue). 

I can submit a pull request.

**Comment by xoofx** [comment]
I have updated the main readme as the status of this project was not clear.

> NOTICE:
>
> This project is no longer actively maintained since 2016, but it's still accepting Pull-Requests.
>
> Small improvements and fixes are welcome, but big refactoring won't likely be.
>
> Please, open an issue in case you are not sure about the changes you want to bring before working on a PR.

**Comment by jbienz** [comment]
Ah, sorry @xoofx I didn't realize the status of the project had changed. I have submitted this fix as PR #114. 

Based on the status of the project (which is understandable), I'm just curious how often you might be able publishing NuGet updates?

Thank you!
